light_prop/lightfield.py

```diff
@@ -0,0 +1,32 @@
+import numpy as np
+
+
+class LightField:
+    def __init__(self, amp, phase):
```

Comment:
Wouldn't `None` and dimension/`type` check here be nice? 

For `type` idk for sure, I'm not really working in untyped lang so I don't know whats the usual thing to do. 

Also checks like that can be moved to `api` layer I proposed in the #4 .

Reply:
Actually we are opting for None - None makes more sense than empty array or array of zeros. We can do the same trick as in `light_prop/propagation_params` or just use [typing](https://docs.python.org/3/library/typing.html)

Reply:
Does `None` make sense if you look at the math operations? Eg how you add/multiply something to.by `None`? Is this the same as `0` or `1`? Do the libs handle eg `fft` with `None` elements? 

The only advantage is memory? Cause you do not keep NxM additional floats? I wouldn't worry about memory for now +  for me it only introduces more problems